packages/go/cypher/model/pgsql/fold/extraction_test.go

```diff
@@ -0,0 +1,70 @@
+package fold_test
+
+import (
+	"github.com/specterops/bloodhound/cypher/frontend"
+	"github.com/specterops/bloodhound/cypher/model/pgsql/fold"
+	"github.com/specterops/bloodhound/cypher/model/pgsql/format"
+	"github.com/specterops/bloodhound/cypher/model/pgsql/translate"
+	"github.com/specterops/bloodhound/cypher/model/pgsql/visualization"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+type TestCase struct {
+	Cypher                 string
+	ExpectedSQLExpressions []string
+	Skip                   bool
+}
+
+func TestExtract(t *testing.T) {
+	testCases := []TestCase{{
+		Cypher: "match (s), (e) where s.name = 'test' and e.name = 'test' and s.value = e.value return s, e",
+		ExpectedSQLExpressions: []string{
+			"s.properties -> 'name' = 'test'",
+			"e.properties -> 'name' = 'test'",
+			"s.properties -> 'value' = e.properties -> 'value'",
+		},
+		Skip: true,
+	}, {
+		Cypher: "match (s), (e) where s.name = 'test' and e.name = 'test' and s.value = e.value return s, e",
+		ExpectedSQLExpressions: []string{
+			"s.properties -> 'name' = 'test'",
+			"e.properties -> 'name' = 'test'",
+			"s.properties -> 'value' = e.properties -> 'value'",
+		},
+	}}
+
+	for _, testCase := range testCases {
+		if testCase.Skip {
+			continue
+		}
+
+		regularQuery, err := frontend.ParseCypher(frontend.NewContext(), testCase.Cypher)
+		require.Nil(t, err)
+
+		sqlAST, err := translate.TranslateCypherExpression(regularQuery.SingleQuery.SinglePartQuery.ReadingClauses[0].Match.Where.Expressions[0])
+		require.Nil(t, err)
+
+		visualization.MustWritePUML(sqlAST, "/home/zinic/digraphs/graph.puml")
```

Comment:
Boo... don't hardcode paths.

Reply:
Cheaper than taking time away from the problem I was focused on at the time. The visualizer needs some love in making it into a proper tool anyway but will address this in more detail once translation is in a better place.